governance/program/src/state.rs

```diff
@@ -0,0 +1,225 @@
+//! Program accounts
+
+use solana_program::pubkey::Pubkey;
+
+/// Max length of a governance name
+pub const MAX_GOVERNANCE_NAME_LENGTH: usize = 32;
+
+/// Max length of a proposal description link
+pub const MAX_PROPOSAL_DESCRIPTION_LINK_LENGTH: usize = 200;
+
+/// Max length of a proposal name
+pub const MAX_PROPOSAL_NAME_LENGTH: usize = 32;
+
+/// Max number of transactions allowed for a proposal
+pub const MAX_TRANSACTIONS: usize = 5;
+
+/// Defines all Governance accounts types
+#[derive(Clone, Debug, PartialEq)]
+pub enum GovernanceAccountType {
+    /// 0 - Default uninitialized account state
+    Uninitialized,
+
+    /// 1 - Governance account
+    Governance,
+
+    /// 2 - Proposal account for Governance account. A single Governance account can have multiple Proposal accounts
+    Proposal,
+
+    /// 3 - Proposal voting state account. Every Proposal account has exactly one ProposalState account
+    ProposalState,
+
+    /// 4 - Vote record account for a given Proposal.  Proposal can have 0..n voting records
+    VoteRecord,
```

Comment:
Why not vote with SPL Tokens, like https://spl.solana.com/feature-proposal does?  Feature Proposal Program only contains a Yes address, but here there could be a Yes and No address

Reply:
Both `Governance` and `Council` mints are for SPL tokens and either one can be selected to vote on a `Proposal` (I explained `Council` in more details in https://github.com/solana-labs/solana-program-library/pull/1635#issuecomment-831504872)

When voters vote they deposit their governance tokens and the amount of deposited tokens is used as their vote weight. The tokens are stored in `Source Token Holding` account while a proposal is being voted on. 

We could use the feature-proposal like approach but that could make creating proposals very expensive (based on the voting token holders population) because for every `Proposal` we would have to create an account with vote tokens for every community member regardless if they would eventually vote or not.

Please note in the `Governance` program  prototype the source tokens are stored per `Proposal` but I'm actually thinking we should store them per top level `Governance`. It would mean you can deposit your source tokens once and then vote with that weight on any number of proposals.

Reply:
Oh got it.  Thanks